ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryBuilder.java

```diff
@@ -650,4 +671,131 @@ private static void updateListProperty(
     valueList.add(value);
     properties.put(key, valueList);
   }
+
+  private static class TransientQuerySinkProcessor implements
```

Comment:
This is complex enough, we might want to pull it out into its own class.

Reply:
Great idea.